.github/workflows/test.yml

```diff
@@ -16,10 +16,11 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip'
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install .
+          pip install -r requirements.txt
```

Comment:
This is good to cleanup s3cmd, but maybe you could still keep an additional line with:
`pip install .`
Because it will call the setup.py of s3cmd and ensure that it is working well also.

Reply:
Sure, I agree. 
Updated